src/DataStructures/Mesh.hpp

```diff
@@ -159,6 +159,25 @@ class Mesh {
   template <size_t N = Dim, Requires<(N > 0 and N == Dim)> = nullptr>  // NOLINT
   Mesh<Dim - 1> slice_away(size_t d) const noexcept;
 
+  /*!
+   * \brief Returns a mesh with only the dimensions `d, ...` present.
+   */
+  template <typename... D, Requires<(sizeof...(D) <= Dim)> = nullptr>
```

Comment:
Is it not possible to take a set of `size_t...`? I know Tensor doesn't do this, but I cannot remember the exact reason why, likely to avoid implicit conversions in generic code. However, that seems like much less of an issue now and especially here, since indexing dimensions we've always done `size_t`. If that's not possible the function should assert that `tt::is_integer_v<D>`

Reply:
I thought "a variable number of arguments of known type" could not be expressed in the C++ template language.  I've wanted it many times.  (But I agree with asserting on the types.)

Reply:
Ah yea, you can have a non-type parameter pack, but that doesn't help us :(
Let's add a static assertion that the types are integer :)